Dockerfile.dind

```diff
@@ -1,9 +1,35 @@
 FROM docker:1.12.3-experimental-dind
 
-RUN apk add --no-cache git tmux py-pip apache2-utils vim \
-    && pip install 'docker-compose==1.8.0'
+ENV DOCKER_STORAGE_DRIVER=overlay2
 
-COPY vimrc ./root/.vimrc
+RUN apk add --no-cache git tmux py-pip apache2-utils vim build-base gettext-dev curl
 
-CMD docker daemon --host=unix:///var/run/docker.sock --host=tcp://0.0.0.0:2375 --storage-driver=vfs &>- & while true; do /bin/sh; done
+# Install Compose and Machine
+RUN pip install docker-compose==1.8.1
+RUN curl -L https://github.com/docker/machine/releases/download/v0.9.0-rc1/docker-machine-Linux-x86_64 \
```

Comment:
Yes, it could work; it will require a bit of cooperation from PWD (for instance, numbering the DinD instances) but it's totally possible!